templates/mq-template.yml

```diff
@@ -0,0 +1,132 @@
+---
+apiVersion: v1
+kind: Template
+metadata:
+  annotations:
+    description: Mosquitto mqtt broker for use with the maestro
+    iconClass: icon-mosquitto
+    openshift.io/display-name: Example Mosquitto MQTT Broker
+    tags: mqtt,mosquitto
+  name: maestro-mqtt-broker
+parameters:
+  - name: MQTT_BROKER_NAME
+    description: The name of the OpenShift Service exposed for the database.
+    displayName: Database Service Name
+    required: true
+    value: maestro-mqtt-broker
+
+  - name: MOSQUITTO_IMAGESTREAM_NAMESPACE
+    description: The OpenShift Namespace where the ImageStream resides.
+    displayName: Namespace
+    value: openshift
+
+  # - name: MQTT_USER
+  #   description: Username for Mosquitto user that will be used for accessing the mqtt broker.
+  #   displayName: Mosquitto Username
+  #   required: false
+  #   value: ocm
+
+  # - name: MQTT_PASSWORD
+  #   description: Password for the Mosquitto connection user.
+  #   displayName: Mosquitto Password
+  #   required: false
+  #   value: TheBlurstOfTimes
+
+  - name: MOSQUITTO_VERSION
+    description: Version of Mosquitto image to be used (2.0.18 or latest).
+    displayName: Version of Mosquitto Image
+    required: true
+    value: "2.0.18"
+
+objects:
+  - apiVersion: v1
+    kind: Service
+    metadata:
+      annotations:
+        template.openshift.io/expose-uri: tcp://{.spec.clusterIP}:{.spec.ports[?(.name==\mosquitto\)].port}
+      name: ${MQTT_BROKER_NAME}
+    spec:
+      ports:
+      - name: mosquitto
+        nodePort: 0
+        protocol: TCP
+        port: 1883
+        targetPort: 1883
+      selector:
+        name: ${MQTT_BROKER_NAME}
+      sessionAffinity: None
+      type: ClusterIP
+    status:
+      loadBalancer: {}
+
+  - apiVersion: apps/v1
+    kind: DeploymentConfig
```

Comment:
a Deployment would be portable to *KS, whereas Deploymentconfig would be openshift only.

Reply:
actually, DeploymentConfig is now deprecated:  https://docs.openshift.com/container-platform/4.14/release_notes/ocp-4-14-release-notes.html#ocp-4-14-deprecated-removed-features

Reply:
replaced `DeploymentConfig` with `Deployment`.